From: Hari Bathini <hbathini@linux.ibm.com>
Date: Wed Mar 16 16:03:05 2022 +0530
Subject: kexec-tools: print error if kexec_file_load fails
References: bsc#1197176
Git-commit: 1d7a308bf7349fcf1627e950159029dfccf85891
Upstream: merged

Commit 4f77da634035 ("kexec-tools: Fix kexec_file_load(2) error
handling") introduced EFALLBACK for scenarios where fallbacking back
to kexec_load syscall is likely to work and dropped printing error
message for these scenarios. But printing error message for other
failure scenarios was inadvertently dropped. Restore printing error
message for such cases.

Fixes: 4f77da634035 ("kexec-tools: Fix kexec_file_load(2) error handling")
Cc: Petr Tesarik <ptesarik@suse.com>
Reported-by: Nageswara R Sastry <rnsastry@linux.ibm.com>
Tested-by: Nageswara R Sastry <rnsastry@linux.ibm.com>
Signed-off-by: Hari Bathini <hbathini@linux.ibm.com>
Reviewed-by: Petr Tesarik <ptesarik@suse.com>
Signed-off-by: Simon Horman <horms@verge.net.au>

---
 kexec/kexec.c |    1 +
 1 file changed, 1 insertion(+)

--- a/kexec/kexec.c
+++ b/kexec/kexec.c
@@ -1294,6 +1294,7 @@ static int do_kexec_file_load(int filein
 		case EMSGSIZE:
 			/* Reject by default. */
 		default:
+			fprintf(stderr, "kexec_file_load failed: %s\n", strerror(errno));
 			ret = EFAILED;
 			break;
 
